src/main/java/net/barrage/school/java/ecatalog/web/ProductController.java

```diff
@@ -33,7 +33,10 @@ public List<Product> listProducts() {
     public List<Product> searchProducts(
             @RequestParam("q") String query
     ) {
-        // TODO Implement a search method which filters products by having q in it's name or description
-        throw new UnsupportedOperationException("Search is not yet implemented :(");
+        var products = productService.listProducts().stream()
```

Comment:
Do u think it's correct place for filtering logic? Why not in the app layer?

minor: NPE in `getDescription().toLowerCase()`. Who tells u ti always will have value?)